### PR TITLE
Update to latest egui and miniquad

### PR DESCRIPTION
**Pull request by emilk**
EDIT: it works well now. Will look into keyboard input tomorrow.

### CODE REVIEW

**Comment by not-fl3** [comment]
Reproduced, and for me this 
```
     pub fn meta() -> ShaderMeta {
         ShaderMeta {
-            images: vec![],
+            images: vec!["u_sampler".to_string()],
             uniforms: UniformBlockLayout {
                 uniforms: vec![UniformDesc::new("u_screen_size", UniformType::Float2)],
             },
```

fixed it, got the UI!

image

**Comment by emilk** [comment]
Well spotted @not-fl3 ! Now I just gotta fix the blur too.

**Comment by emilk** [comment]
`high_dpi` blocked by https://github.com/not-fl3/miniquad/issues/169

File: examples/test_window.rs [hunk 1]
```diff
@@ -6,17 +6,43 @@ use {
...
+            screen_rect: Some(egui::Rect::from_min_size(Default::default(), screen_size)),
+            pixels_per_point: Some(pixels_per_point),
+            ..Default::default()
+        };
+
+        Self {
+            egui_ctx,
+            painter: Painter::new(ctx),
+            raw_input,
+            start_time: Instant::now(),
```
Comment (not-fl3) [inline examples/test_window.rs:33]: Note that `Instant` do not really work on wasm. 
Alternatives are:
`miniquad::now()` - will give time as f64
or https://github.com/sebcrozet/instant - uses `miniquad::now` under the hood and will give a struct similar to time::Instant

File: src/painter.rs [hunk 2]
```diff
@@ -130,109 +128,145 @@ impl Painter {
...
-                uv: x.uv,
-                color: (x.color.0[0], x.color.0[1], x.color.0[2], x.color.0[3]),
-            })
-            .collect::<Vec<Vertex>>();
-        self.bindings.vertex_buffers[0].update(ctx, &vertices);
+        self.bindings.vertex_buffers[0].update(ctx, &mesh.vertices);
+
+        let screen_size = ctx.screen_size();
 
         // TODO: support u32 indices in miniquad and just use "mesh.indices"
```
Comment (not-fl3) [inline src/painter.rs:135]: If you did not looked into this yet - I can do it

Comment (emilk) [reply src/painter.rs:135]: I added a call to the [`split_to_u16()`](https://docs.rs/epaint/0.8.0/epaint/struct.Triangles.html#method.split_to_u16) so 32-bit indices is not a very high priority. The only thing that could break if is somebody actually adds a triangle mesh to the egui painter that uses more than 2^16 vertices, but normal egui usage will be fine.

**Review by not-fl3 — APPROVED** [review]

**Comment by emilk** [comment]
This works rather well now!

Still TODO (for this or later PRs):

* [ ] Copy/paste (can use the `clipboard` for that)
* [ ] Hyperlinks (can use the `webbrowser` for that)
* [ ] Touch input
* [ ] Test on web
* [ ] Wrap the required state (egui context, egui input and the painter) in a `struct` for easier reuse
* [ ] Rename crate `egui_miniquad` and publish to crates.io

Then all that is really lacking is [some way to set a cursor in miniquad](https://github.com/not-fl3/miniquad/issues/171)

**Comment by not-fl3** [comment]
>  Copy/paste (can use the clipboard crate for that)

`miniquad` itself has clipboard support. Unfortunately it is implemented on web, windows and linux but not mac :( 

https://github.com/not-fl3/miniquad/blob/master/src/clipboard.rs#L60

**Pull request merged**